src/opustrainer/trainer.py

```diff
@@ -204,7 +204,7 @@ def _read_line(self) -> None:
         try:
             # Try to find the next non-empty line
             while True:
-                self._next_line = self._fh.readline()
+                self._next_line = self._fh.readline() # type: ignore _fh can't be none.
```

Comment:
A neater way is adding an `assert self._fh is not None` somewhere above these lines. Then your type checker should be happy.

Reply:
I tried that, but this one is very obvious that it can't be None, so I decided to use the Pylance ignore pragma